### PR TITLE
Hacky checksum fail

### LINKED ISSUE #524: Checksum failed for nightly channel
I just moved from multirust to rustup and experience the following problem when installing nightly:
```
$> rustup update nightly
info: syncing channel updates for 'nightly-x86_64-unknown-linux-gnu'
error: checksum failed, expected: 'afb2094f716fb7167accccdbc9e213ad5e707e7b56b0175aa664108cf994eaa0', calculated: 'b60f814a446940366499a350769dab93bce25e91bb30825eebb3f74820fd8b74'
```

Stable and beta install just fine. Also I removed all remains of multirust from my system(removed ~/.multirust, cleared $PATH, fresh console) and tried again, no help.

To clarify, I do not have any version of nightly currently installed (it seems to work for people with nightly already installed):
```
$> rustup show
installed toolchains
--------------------

stable-x86_64-unknown-linux-gnu (default)
beta-x86_64-unknown-linux-gnu

active toolchain
----------------

stable-x86_64-unknown-linux-gnu (default)
rustc 1.9.0 (e4e8b6668 2016-05-18)
```

I am using the current version of rustup:
```
$> rustup -V
rustup 0.1.12 (c6e430a 2016-05-12)
```

### PR DESCRIPTION
**Pull request by AtheMathmo**
Resolved #524 (sort of...)

The implementation is also worryingly hacky.. Sorry. Please feel free to suggest a cleaner implementation if you see one.

Also it looks like I changed the formatting in places - if that is an issue let me know and I will correct it. Changes to `dist.rs` [begin here](https://github.com/rust-lang-nursery/rustup.rs/compare/master...AtheMathmo:hacky-checksum-fail?expand=1#diff-30760ac1a65d86b9a5bc9200ec773c75R560)

### CODE REVIEW

File: src/rustup-dist/src/dist.rs [hunk 1]
```diff
@@ -539,13 +539,25 @@ fn dl_v2_manifest<'a>(download: DownloadCfg<'a>,
...
+        let (manifest_file, manifest_hash) = if let Some(m) = manifest_dl { m } else { return Ok(None) };
+        let manifest_str = try!(utils::read_file("manifest", &manifest_file));
+        let manifest = try!(ManifestV2::parse(&manifest_str));
 
-    Ok(Some((manifest, manifest_hash)))
+        Ok(Some((manifest, manifest_hash)))
+    } else {
+        match *manifest_dl_res.as_ref().unwrap_err().kind() {
+            // Checksum failed - issue warning to try again later
+            ErrorKind::ChecksumFailed{url: _, expected: _, calculated: _} => (download.notify_handler)(Notification::ManifestChecksumFailedHack),
```
Comment (brson) [inline src/rustup-dist/src/dist.rs]: Please replace these three wildcard field patterns with `{ .. }`.

**Comment by brson** [comment]
Looks good. Just that one nit to address in the verbose pattern. Thanks!

**Comment by AtheMathmo** [comment]
Thanks, have made your suggested change. (I totally forgot I could pattern match like that :D )

**Comment by brson** [comment]
Thanks!

**Pull request merged**